### PR TITLE
feat(gateway): support custom slash commands

### PR DESCRIPTION
**Pull request by chaychoong**
## What does this PR do?

Scan executable files dropped into ~/.hermes/commands/ at gateway startup and register them as slash commands. Matching commands bypass the LLM loop entirely - the script is executed directly with HERMES_CHAT_ID, HERMES_PLATFORM, and HERMES_ARGS injected as env vars, and its stdout is returned as the response.

Previously, adding a custom quick command required either a source edit or a config entry with a specific schema. A drop-in directory with zero-config auto-discovery is lower friction, and the LLM-bypass is intentional: these are deterministic, token-free invocations — suited for things like /digest or /status where you want instant output without an agent round-trip.

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ New feature (non-breaking change that adds functionality)
- [ ] 🔒 Security fix
- [ ] 📝 Documentation update
- [ ] ✅ Tests (adding or improving test coverage)
- [ ] ♻️ Refactor (no behavior change)
- [ ] 🎯 New skill (bundled or hub)

## How to Test

1. Ask the agent to create a custom command.
2. Trigger the command on whatever channel you use (I tested on Telegram)

### Code

- [x] I've read the [Contributing Guide](https://github.com/NousResearch/hermes-agent/blob/main/CONTRIBUTING.md)
- [x] My commit messages follow [Conventional Commits](https://www.conventionalcommits.org/) (`fix(scope):`, `feat(scope):`, etc.)
- [x] I searched for [existing PRs](https://github.com/NousResearch/hermes-agent/pulls) to make sure this isn't a duplicate
- [x] My PR contains **only** changes related to this fix/feature (no unrelated commits)
- [x] I've run `pytest tests/ -q` and all tests pass
- [x] I've added tests for my changes (required for bug fixes, strongly encouraged for features)
- [x] I've tested on my platform: 

### Documentation & Housekeeping

- [ ] I've updated relevant documentation (README, `docs/`, docstrings) — or N/A
- [ ] I've updated `cli-config.yaml.example` if I added/changed config keys — or N/A
- [ ] I've updated `CONTRIBUTING.md` or `AGENTS.md` if I changed architecture or workflows — or N/A
- [ ] I've considered cross-platform impact (Windows, macOS) per the [compatibility guide](https://github.com/NousResearch/hermes-agent/blob/main/CONTRIBUTING.md#cross-platform-compatibility) — or N/A
- [ ] I've updated tool descriptions/schemas if I changed tool behavior — or N/A

### CODE REVIEW

**Comment by juliolazzarim** [comment]
Hey @chaychoong — this would unblock a real use case on my end. I run a local Hermes setup on WSL2 and have been carrying a custom `close-session` skill that wraps the FASE 3 lifecycle (drift check, current_focus, TODO, handoff, report). Today the only way to trigger it is a natural-language phrase like "fecha a sessão" — works, but `/close-session` is what I keep typing by reflex and it errors with `not a quick/plugin/skill command`.

Two things from reading the diff that may help reviewers:

1. **Path resolution cross-platform.** `~/.hermes/commands/` assumes `$HOME` is set and that the gateway process inherits it. On systemd-managed setups I use `User=juliotta` for `bdex-web.service` — `$HOME` is still set, but I've seen edge cases on Windows-native installs where it points to a different shell home. Probably worth a `Path.home()` fallback with a config override.

2. **Error handling when a command exits non-zero.** I couldn't see in the 47 lines you added in `hermes_cli/commands.py` what happens if the script returns exit 1 or hangs. A non-zero exit silently swallowed is worse than no command at all — the user typed `/close-session`, nothing happened, they retry, and the drift check runs twice. A `stderr → reply` path with a 5-10s timeout would be the safe default.

Happy to test the branch on my side once you push a fix for either — I can run the close-session flow end-to-end and report back. Or if it's useful, I can open a tracking issue for the Windows `$HOME` edge case.

Thanks for picking the drop-in shape over the registry schema from #25335. Zero-config beats another config file every time.

File: gateway/run.py [hunk 1]
```diff
@@ -1984,13 +1988,38 @@ async def _handle_message(self, event: MessageEvent) -> Optional[str]:
...
         if command:
             if isinstance(self.config, dict):
                 quick_commands = self.config.get("quick_commands", {}) or {}
             else:
                 quick_commands = getattr(self.config, "quick_commands", {}) or {}
             if not isinstance(quick_commands, dict):
                 quick_commands = {}
+
+            # Check user script commands first
+            if canonical in self._user_commands or command in self._user_commands:
```
Comment (teknium1) [inline gateway/run.py:2001]: This starts from the gateway's full environment, which contains provider and platform credentials. Current main intentionally sanitizes quick-command environments in `gateway/run.py:9999-10008`; preserve that protection and add only the three command metadata variables.

File: gateway/run.py [hunk 2]
```diff
@@ -1984,13 +1988,38 @@ async def _handle_message(self, event: MessageEvent) -> Optional[str]:
...
+                script_path = self._user_commands.get(canonical) or self._user_commands.get(command)
+                args = event.text.strip()[len(f"/{command}"):].strip() if event.text else ""
+                env = {**__import__("os").environ,
+                       "HERMES_CHAT_ID": str(event.source.chat_id),
+                       "HERMES_PLATFORM": str(event.source.platform.value if hasattr(event.source.platform, "value") else event.source.platform),
```
Comment (teknium1) [inline gateway/run.py:2006]: `args.split()` breaks quoted arguments such as `/digest "weekly report"`. Use shell-aware parsing before constructing the argv.

File: gateway/run.py [hunk 3]
```diff
@@ -1984,13 +1988,38 @@ async def _handle_message(self, event: MessageEvent) -> Optional[str]:
...
+                       "HERMES_ARGS": args}
+                try:
+                    proc = await asyncio.create_subprocess_exec(
+                        script_path, *args.split() if args else [],
+                        stdout=asyncio.subprocess.PIPE,
+                        stderr=asyncio.subprocess.PIPE,
```
Comment (teknium1) [inline gateway/run.py:2012]: Check `proc.returncode` and return a distinct failure response for non-zero exits; otherwise stderr from a failed command is returned as if it were a successful command result.

**Review by teknium1 — COMMENTED** [review]
Thanks for the concrete zero-token gateway-command use case. The drop-in discovery feature is not on current main, but this patch needs rework before it can safely reuse the gateway command path.

### Problems
- Proposed `gateway/run.py:2001-2013` passes the full gateway environment to the script and returns output without redaction. Current main intentionally sanitizes quick-command environments at `gateway/run.py:9999-10008` and redacts output at `gateway/run.py:10012-10015`; commit `f6736ced8123e4e17bc0bde89b208c0baedbf0c4` fixed credential leakage in this exact path.
- Proposed `hermes_cli/commands.py:206-214` still adds colliding names to the executable dispatch map even when registry registration is skipped. Reserve built-in names before insertion, rather than only suppressing registration.
- Proposed `gateway/run.py:2006` uses `args.split()`, which does not preserve quoted arguments, and the proposed execution path does not inspect `returncode`. This matches the non-zero-exit concern in the existing review comment.
- Please add coverage comparable to `tests/cli/test_quick_commands.py:164-200` and document the new persistent command directory alongside `website/docs/user-guide/configuration.md:1674-1705`.

### Suggested changes
- Reuse the current environment-sanitization and output-redaction protections while explicitly injecting only the intended command metadata.
- Define collision precedence, use shell-aware argument parsing, and surface non-zero exits distinctly.

Automated hermes-sweeper review.

File: hermes_cli/commands.py [hunk 4]
```diff
@@ -173,6 +173,53 @@ def register_plugin_command(cmd: CommandDef) -> None:
...
+        return user_commands
+
+    existing_names = {cmd.name for cmd in COMMAND_REGISTRY}
+
+    for script in sorted(cmds_path.iterdir()):
+        if not script.is_file():
+            continue
+        if not os.access(script, os.X_OK):
+            continue
+        # Use stem (filename without extension) as command name
```
Comment (teknium1) [inline hermes_cli/commands.py:206]: The collision check below only skips registry registration, but this assignment still makes a colliding filename executable at dispatch. Reserve existing command names before inserting into `user_commands`, or document an explicit override policy.
